file.go

```diff
@@ -90,6 +96,26 @@ var kindStr = map[yaml.Kind]string{
 	yaml.AliasNode:    "AliasNode",
 }
 
+// NewJsonVisitor returns a visitor object that satisfies the Queryable interface
+// attempting to turn a supposed JSON byte slice into a *yaml.Node object
+func NewJsonVisitor(buf []byte) (Queryable, error) {
+	visitor := &yamlVisitor{
```

Comment:
I like the term `visitor`. Is that yours?

Reply:
it's a term I learned from `sqlx`: https://en.wikipedia.org/wiki/Visitor_pattern